### ISSUE TITLE
seg fault training tf.nn.conv3d with minibatch size >2

### ISSUE THREAD

**Issue by djbard** [tensorflow/tensorflow#14807]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Custom code! You can find it here: https://github.com/NERSC/CosmoFlow/tree/master/SegFault
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: SUSE Linux 12.2
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: ('v1.3.0-rc1-3112-g65b6a75', '1.4.0-rc0') Note this is NOT compiled with the Intel MKL options. 
- **Python version**: 2.7.13 
- **Bazel version (if compiling from source)**: 0.6.0
- **GCC/Compiler version (if compiling from source)**: 4.8.5
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A, Running on x86_64 Intel Haswell node
- **Exact command to reproduce**: See README in https://github.com/NERSC/CosmoFlow/tree/master/SegFault

### Describe the problem

A seg fault when training a tf.nn.conv3d with minibatch size more than 2 on a single Intel Haswell. The seg fault occurs at [line 187](https://github.com/NERSC/CosmoFlow/blob/master/SegFault/CosmoNet.py#L187).  

### Source code / logs

GDB log: https://github.com/NERSC/CosmoFlow/blob/master/SegFault/gdbTrace.log
It looks like some kind of cyclic dependency in Eigen::TensorEvaluator.

**Comment by facaiy** [comment]
It could be more helpful if you could create a mini reproducable example. Thanks.

**Comment by djbard** [comment]
I agree - but I haven't had time to make a mini example (this is someone else's code). I will try to find time to do so, but in the meantime the github repo I linked to will reproduce the error shown in the gdb log.

**Comment by tatatodd** [comment]
@djbard Thanks for filing the issue!  As @facaiy mentioned, a minimum repro would be appreciated.  But I agree that this looks like a cycle in Eigen that ends up exhausting the stack.

@benoitsteiner @rmlarsen Can you take a look, or suggest someone who can?

Here's a simplified form of the bottom of the stack trace from the `gdbTrace.log` link above.  I dunno if this is simply a bug that causes an infinite loop, or if there's some static unrolling that's managing to blow our stack, but having the same repeated pattern of calls for >7000 frames definitely looks suspicious:
```
... pattern repeats until frame #0 which segfaults ...
#7671 Eigen::TensorEvaluator::Context::kernel(long, long, long) ()
  
#7672 Eigen::TensorEvaluator::Context::signal_kernel(long, long, long, bool) ()
#7673 Eigen::TensorEvaluator::Context::pack_rhs(long, long) ()
#7674 Eigen::TensorEvaluator::Context::enqueue_packing_helper(long, long, long, bool) ()
#7675 Eigen::TensorEvaluator::Context::pack_lhs(long, long) ()
#7676 Eigen::TensorEvaluator::Context::enqueue_packing_helper(long, long, long, bool) ()
#7677 Eigen::TensorEvaluator::Context::kernel(long, long, long) ()
  
#7678 Eigen::TensorEvaluator::Context::signal_kernel(long, long, long, bool) ()
#7679 Eigen::TensorEvaluator::Context::pack_rhs(long, long) ()
#7680 Eigen::TensorEvaluator::Context::enqueue_packing_helper(long, long, long, bool) ()
#7681 Eigen::TensorEvaluator::Context::pack_lhs(long, long) ()
#7682 Eigen::TensorEvaluator::Context::enqueue_packing_helper(long, long, long, bool) ()
#7683 Eigen::TensorEvaluator::Context::kernel(long, long, long) ()
```
For readability, I simplified `Eigen::TensorEvaluator::Context` above from the full form, in all its templatized glory:
```
Eigen::TensorEvaluator<Eigen::TensorContractionOp<Eigen::array<Eigen::IndexPair<long>, 1ul> const, Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<float const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<float const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>::Context<Eigen::internal::gemm_pack_lhs<float, long, Eigen::internal::TensorContractionSubMapper<float, long, 1, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<float const, 5, 1, long>, 16, Eigen::MakePointer> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 4, true, false, 0, Eigen::MakePointer>, 8, 4, 0, false, false>, Eigen::internal::gemm_pack_rhs<float, long, Eigen::internal::TensorContractionSubMapper<float, long, 0, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<float const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 4, true, false, 0, Eigen::MakePointer>, 4, 0, false, false>, Eigen::internal::gebp_kernel<float, float, long, Eigen::internal::blas_data_mapper<float, long, 0, 0>, 8, 4, false, false>, Eigen::internal::TensorContractionInputMapper<float, long, 1, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<float const, 5, 1, long>, 16, Eigen::MakePointer> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 4, true, false, 0, Eigen::MakePointer>, Eigen::internal::TensorContractionInputMapper<float, long, 0, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<float const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 4, true, false, 0, Eigen::MakePointer>, Eigen::internal::blas_data_mapper<float, long, 0, 0> >
```

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by djbard** [comment]
Has anyone had a chance to look at this issue @tatatodd?

**Comment by tatatodd** [comment]
I dunno - @benoitsteiner @rmlarsen can you take a look, or suggest someone who can?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by reedwm** [comment]
This looks like a duplicated of  #16139 which has an example to reproduce, so I'm closing this issue.
